package/MDAnalysis/analysis/base.py

```diff
@@ -170,12 +175,15 @@ def run(self, start=None, stop=None, step=None, verbose=None):
             number of frames to skip between each analysed frame
         verbose : bool, optional
             Turn on verbosity
+            
+        .. versionchanged:: 1.0.0
+            Verbose argument is now passed explicitly to
+            :meth:`AnalysisBase._setup_frames`.
         """
         logger.info("Choosing frames to analyze")
         # if verbose unchanged, use class default
         verbose = getattr(self, '_verbose', False) if verbose is None else verbose
-
-        self._setup_frames(self._trajectory, start, stop, step)
+        self._setup_frames(self._trajectory, start, stop, step, verbose)
```

Comment:
I don't think this is correct. Why should a method of a class receive an argument that is an attribute of that same class when the method being called is not a `@staticmethod` ? (may be this is not from this PR and I am missing something)

Reply:
`start`, `stop`, `step`, and `verbose` are historically argument of the class, though we are deprecating them from the initializer in favour of the same arguments in the `run` method.

Reply:
Sorry, I was referring to the `self._trajectory`. Because it also gets assigned again on `_setup_frames`. But this is for other PR/issue, indeed.